sidebar.c

```diff
@@ -903,15 +902,111 @@ static bool imap_is_prefix(const char *folder, const char *mbox, size_t *plen)
   if (mutt_str_strncmp(url_m->path, url_f->path, flen) != 0)
     goto done;
 
-  if (url_m->user && !url_f->user)
-    *plen += mutt_str_strlen(url_m->user) + 1;
-  rc = true;
+  plen = strlen(mbox) - mlen + flen;
 
 done:
   url_free(&url_m);
   url_free(&url_f);
 
-  return rc;
+  return plen;
+}
+
+/**
+ * abbrev_folder - Abbreviate a Mailbox path using a folder
+ * @param mbox   Mailbox path to shorten
+ * @param folder Folder path to use
+ * @param type   Mailbox type
+ * @retval ptr Pointer into the mbox param
+ */
+static const char *abbrev_folder(const char *mbox, const char *folder, enum MailboxType type)
+{
+  if (!mbox || !folder)
+    return NULL;
+
+  if (type == MUTT_IMAP)
+  {
+    int prefix = imap_is_prefix(folder, mbox);
+    if (prefix == 0)
+      return NULL;
+    return mbox + prefix;
+  }
+
+  if (!C_SidebarDelimChars)
+    return NULL;
+
+  size_t flen = mutt_str_strlen(folder);
+  if (flen == 0)
+    return NULL;
+  if (strchr(C_SidebarDelimChars, folder[flen - 1])) // folder ends with a delimiter
+    flen--;
+
+  size_t mlen = mutt_str_strlen(mbox);
+  if (mlen <= flen)
+    return NULL;
+
+  if (mutt_str_strncmp(folder, mbox, flen) != 0)
+    return NULL;
+
+  // After the match, check that mbox has a delimiter
+  if (!strchr(C_SidebarDelimChars, mbox[flen]))
+    return NULL;
+
+  return mbox + flen + 1;
+}
+
+/**
+ * abbrev_net - Abbreviate a network Mailbox path
+ * @param mbox Mailbox path to shorten
+ * @param type Mailbox type
+ *
+ * Use heuristics to shorten a non-local Mailbox path.
+ * Strip the host part (or database part for Notmuch).
+ */
+static const char *abbrev_net(const char *mbox, enum MailboxType type)
```

Comment:
I also think that `abbrev_net` could use some clarification. Isn't it amounting to returning the path+query part of a URL?

Reply:
I've renamed it to `abbrev_url()` and added some examples.